wls-gui-wahllokalsystem/tests/unit/fetch-utils.spec.ts

```diff
@@ -0,0 +1,109 @@
+
+import { wlsCatchHandler, wlsResponseHandler } from "@/api/fetch-utils";
+
+// Da `vi.mock` vor allen Imports ausgeführt wird, gibt es die Fehlermeldung "Error: [vitest] There was an error when
+// mocking a module. If you are using "vi.mock" factory, make sure there are no top level variables inside, since this
+// call is hoisted to top of the file. Read more: https://vitest.dev/api/vi.html#vi-mock"." Um diese zu umgehen, werden
+// die gemockten Funktionen in `vi.hoisted` gelistet.
+const { mockCreateDefaultWlsError } = vi.hoisted(() => ({
+  mockCreateDefaultWlsError: vi.fn(),
+}));
+
+vi.mock("@/api/WLSError", () => ({
+  createDefaultWlsError: mockCreateDefaultWlsError,
+}));
+
+describe("fetch-utils.ts", () => {
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  describe("wlsResponseHandler", () => {
+    // Parametrized Test, um verschiedene Szenarien (response codes) zu testen. Mit `$variablenname` kann der
+    // entsprechende Parameter (response code) mit im Testnamen aufgenommen werden.
+    it.each([
+      { response: new Response(null, { status: 204 }) },
+      { response: new Response('{"error": "unauthorized"}', { status: 401 }) },
+      { response: new Response('{"error": "not found"}', { status: 404 }) },
+      { response: new Response('{"error": "bad request"}', { status: 400 }) },
+      {
+        response: new Response('{"error": "internal server error"}', {
+          status: 500,
+        }),
+      },
+    ])(
+      "should_returnRejectedResponse_when_responseCodeIs$response.status",
+      async ({ response }) => {
+        await expect(wlsResponseHandler(response)).rejects.toEqual(response);
+      }
+    );
+
+    it("should_returnResolvedResponse_when_responseCodeIs200", async () => {
+      const response = new Response('{"data": "test"}', { status: 200 });
+      await expect(wlsResponseHandler(response)).resolves.toEqual(response);
+    });
+  });
+
+  describe("wlsCatchHandler", () => {
+    it("should_throwWlsErrorWithNoContentMessage_when_responseCodeIs204", async () => {
+      const mockedResponse = new Response(null, { status: 204 });
+      const errorMessage = "Es konnten keine Daten gefunden werden";
+
+      // definiert das Verhalten der Methode
+      mockCreateDefaultWlsError.mockReturnValue(new Error(errorMessage));
+
+      // `expect(wlsCatchHandler()).toThrow` ohne Arrow-Funktion wirft den Fehler: "Error: Es konnten keine Daten
+      // gefunden werden", weil die Funktion tatsächlich ausgeführt wird. Mit der Kapselung in die Arrow-Funktion kann
+      // der Test vorher erkennen, dass ein Fehler geworfen wird. Das expect-Statement führt dann die Funktion aus und
+      // überprüft, ob die erwartete Exception auftritt.
+      expect(() => wlsCatchHandler(mockedResponse)).toThrow(errorMessage);
+      expect(mockCreateDefaultWlsError).toHaveBeenCalledWith({
+        message: errorMessage,
+        code: mockedResponse.status.toString(),
+      });
```

Comment:
`errorMessage` ist ungünstig weil die ErrorMessage nicht die gleiche sein muss die beim Aufruf von `mockCreateDefaultWlsError` als Argument übermittel wird. Das trifft auch auf die anderen Tests zu.

Mein Verbesserungsvorschlag ist mit einem Regex zu arbeiten innerhalb von `objectContaining`

```suggestion
      expect(mockCreateDefaultWlsError).toHaveBeenCalledWith(
        expect.objectContaining({
          message: expect.stringMatching(/.+/),
          code: mockedResponse.status.toString(),
        })
      );
```

Reply:
beim code 204 ist die message aktuell immer die gleiche, da habe ich es so gelassen. bei den anderen tests habe ich es angepasst. siehe [cf2af46](https://github.com/it-at-m/Wahllokalsystem/pull/702/commits/cf2af46d22449e751443a4da036bb4e37831853a)